### ISSUE TITLE
Distinguishing between explicit null and omitted value when object given as input

### ISSUE THREAD

**Issue by marisofie** [graphql-dotnet/graphql-dotnet#3308]
Hello, 

We are trying to implement mutations that support partial updates, and would like to have objects as input. In our current implementation we distinguish between explicit null and omitted arguments by having an additional parameter `valueSet` per parameter on the incoming object, that is set to true in the object parameters' setter. 

Previously, we have utilized `IResolveFieldContext.hasArgument()` and have had individual input arguments. It would have been great for a similar functionality when utilizing objects as input, but we have not found a way to see if parameters belonging to an object are set as input in our mutation by accessing the `IResolveFieldContext`. 

My question is therefore if I have missed some functionality that already exists, or if it could be possible to extend GraphQL so it is possible to distinguish between explicit null and when the object's parameters are null due to omitted arguments?

**Comment by sungam3r** [comment]
See #3065 #2921

**Comment by Shane32** [comment]
The basis of your request can be found by overriding `ParseDictionary` within your input object.  Only fields that were set will appear in the dictionary, which is according to GraphQL spec.  Then you have to coerce it to an object to return to the field resolver.  An easy option would be to return the dictionary itself, and within your mutation you would call `GetArgument<Dictionary<string, object?>>("value")` or similar to retrieve the raw dictionary.

See also: 
- #575 
- #541

**Comment by sungam3r** [comment]
Do we have documentation about it ?

**Comment by Shane32** [comment]
No, but we should.  If nothing else, in the FAQ, along with a simple sample.

**Comment by marisofie** [comment]
That works perfectly, thank you :)
